modules/PROGENY.py

```diff
@@ -1,234 +1,140 @@
-                                    # TODO check is this the correct evidence code "computational combinatorial evidence",
-                                    evidence_codes = ["http://purl.obolibrary.org/obo/ECO_0000053"],
-                                    provenance_type = provenance_type,
-                                    resource_score = resource_score,
-                                    urls=[linkout]
-                                )
-                                # Add target_id to unique_association_fields
-                                evidenceString.unique_association_fields['target_id'] = evidenceString.target.id
-                                # Append current evidence string
-                                self.evidence_strings.append(evidenceString)
-                            else:
-                                self.logger.error("%s is not found in HGNC" % gene_symbol)
-
-            self.logger.error("%s evidence parsed"%(n-1))
-            self.logger.error("%s evidence created"%len(self.evidence_strings))
-
-    def write_evidence(self, filename=Config.PROGENY_EVIDENCE_FILENAME):
-        self.logger.info("Writing PROGENY evidence strings")
-        with open(filename, 'w') as progeny_output:
-            n = 0
-            for evidence_string in self.evidence_strings:
-                n += 1
-                self.logger.info(evidence_string.disease.id[0])
-                error = evidence_string.validate(logging)
-                if error == 0:
-                    progeny_output.write(evidence_string.to_JSON(indentation=None)+"\n")
-                else:
-                    self.logger.error("REPORTING ERROR %i" %n)
-                    self.logger.error(evidence_string.to_JSON(indentation=4))
-
-def main():
-    PROGENY().process_progeny()
-
-if __name__ == "__main__":
-    main()
-
-
 
+        return self.dataframe
+    
+    def pathway2Reactome(self):
+        pathwayMappingsFile = self.spark \
+                        .read.csv("resources/pathway2Reactome_mappings.tsv", sep=r'\t', header=True) \
+                        .withColumnRenamed("pathway", "Pathway")
+        
+        self.dataframe = self.dataframe \
+                .join(
+                    pathwayMappingsFile,
+                    on="Pathway",
+                    how="inner"
+                ) \
+                .withColumn(
+                    "target",
+                    split(col("target"), ", ") 
+                ) \
+                .withColumn(
+                    "target",
+                    explode("target")
+                )
+
+        return self.dataframe
+
+    @staticmethod
+    def parseEvidenceString(row):
+        try:
+            evidence = {
+                "datasourceId" : "progeny",
+                "datatypeId" : "affected_pathway",
+                "diseaseFromSource" : row["Cancer_type"],
+                "diseaseFromSourceMappedId" : row["EFO_id"],
```

Comment:
What is the value of `row["Cancer_type"]` and `row["EFO_id"]` when `mappingStep` is `False`?

Reply:
The `diseaseFromSource` field will remain the same, if the mappingStep is False only `diseaseFromSourceMappedId` will be affected. I've added the condition that if that were the case, this will be `None` as a safeguard. Until we don't switch off the mappingStep on our side, we don't know what is the best solution to feed the ETL.

Reply:
This approach was improved by creating the field only if the value of `row["EFO_id"]` existed.